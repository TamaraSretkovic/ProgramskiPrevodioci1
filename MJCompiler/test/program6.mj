//	Semantika test 
//  - designator dodele nije var ili elem
//  - kompatibilnost reference sa dodelom null
//  - nekompatibilnost tipova pri dodeli

program test302
	int niz[];
{
	void main() {
		int = niz;
        niz = 'c';
        niz = null;
        niz = new int[5];
	}
}

