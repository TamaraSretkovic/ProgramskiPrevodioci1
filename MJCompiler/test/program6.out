INFO  01:15:54,739 - Compiling source file: /home/tamarica/Documents/tamara/velika skola/8 semestar/PP1/projekat/projecat_2019_2020/MJCompiler/test/program6.mj
INFO  01:15:54,763 - #2 program
INFO  01:15:54,763 - #46 test302
INFO  01:15:54,763 - #46 int
INFO  01:15:54,763 - #46 niz
INFO  01:15:54,764 - #8 [
INFO  01:15:54,764 - #9 ]
INFO  01:15:54,764 - #5 ;
INFO  01:15:54,764 - #3 {
INFO  01:15:54,765 - #10 void
INFO  01:15:54,765 - #46 main
INFO  01:15:54,765 - #12 (
INFO  01:15:54,765 - #11 )
INFO  01:15:54,766 - #3 {
INFO  01:15:54,766 - #46 int
INFO  01:15:54,766 - #23 =
INFO  01:15:54,766 - #46 niz
INFO  01:15:54,767 - #5 ;
INFO  01:15:54,767 - #46 niz
INFO  01:15:54,768 - #23 =
INFO  01:15:54,768 - #48 c
INFO  01:15:54,768 - #5 ;
INFO  01:15:54,768 - #46 niz
INFO  01:15:54,768 - #23 =
INFO  01:15:54,768 - #46 null
INFO  01:15:54,768 - #5 ;
INFO  01:15:54,768 - #46 niz
INFO  01:15:54,768 - #23 =
INFO  01:15:54,768 - #36 new
INFO  01:15:54,768 - #46 int
INFO  01:15:54,769 - #8 [
INFO  01:15:54,769 - #47 5
INFO  01:15:54,769 - #9 ]
INFO  01:15:54,769 - #5 ;
INFO  01:15:54,769 - #4 }
INFO  01:15:54,769 - #4 }
INFO  01:15:54,773 - ==================SINTAKSNO STABLO==================
INFO  01:15:54,774 - Program(
  ProgName(
   test302
  ) [ProgName]
  DeclarationListDeclaration(
    NoDeclarationList(
    ) [NoDeclarationList]
    DeclarationListVar(
      VarDeclarationV(
        VarType(
          Type(
           int
          ) [Type]
        ) [VarType]
        VarDecl(
          IdentArray(
           niz
          ) [IdentArray]
        ) [VarDecl]
      ) [VarDeclarationV]
    ) [DeclarationListVar]
  ) [DeclarationListDeclaration]
  MethodDeclarationListMethod(
    NoMethodDeclarationList(
    ) [NoMethodDeclarationList]
    VoidMethodDeclaration(
      MethodVoidName(
       main
      ) [MethodVoidName]
      MethodDeclarationActual(
        NoFormalParamsList(
        ) [NoFormalParamsList]
        NoDeclarationList(
        ) [NoDeclarationList]
        StatementListStatement(
          StatementListStatement(
            StatementListStatement(
              StatementListStatement(
                NoStatementList(
                ) [NoStatementList]
                DesignatorStatementStatement(
                  LeftDesignatorAssignExpression(
                    LeftSideAssignVar(
                      DesignatorOnly(
                       int
                      ) [DesignatorOnly]
                    ) [LeftSideAssignVar]
                    AssignOpEqual(
                      AssignOp(
                      ) [AssignOp]
                    ) [AssignOpEqual]
                    ExpressionTermDef(
                      OnlyTerm(
                        TermFactor(
                          FactorDesignator(
                            DesignatorOnly(
                             niz
                            ) [DesignatorOnly]
                          ) [FactorDesignator]
                        ) [TermFactor]
                      ) [OnlyTerm]
                    ) [ExpressionTermDef]
                  ) [LeftDesignatorAssignExpression]
                ) [DesignatorStatementStatement]
              ) [StatementListStatement]
              DesignatorStatementStatement(
                LeftDesignatorAssignExpression(
                  LeftSideAssignVar(
                    DesignatorOnly(
                     niz
                    ) [DesignatorOnly]
                  ) [LeftSideAssignVar]
                  AssignOpEqual(
                    AssignOp(
                    ) [AssignOp]
                  ) [AssignOpEqual]
                  ExpressionTermDef(
                    OnlyTerm(
                      TermFactor(
                        FactorConst(
                          CharConst(
                           c
                          ) [CharConst]
                        ) [FactorConst]
                      ) [TermFactor]
                    ) [OnlyTerm]
                  ) [ExpressionTermDef]
                ) [LeftDesignatorAssignExpression]
              ) [DesignatorStatementStatement]
            ) [StatementListStatement]
            DesignatorStatementStatement(
              LeftDesignatorAssignExpression(
                LeftSideAssignVar(
                  DesignatorOnly(
                   niz
                  ) [DesignatorOnly]
                ) [LeftSideAssignVar]
                AssignOpEqual(
                  AssignOp(
                  ) [AssignOp]
                ) [AssignOpEqual]
                ExpressionTermDef(
                  OnlyTerm(
                    TermFactor(
                      FactorDesignator(
                        DesignatorOnly(
                         null
                        ) [DesignatorOnly]
                      ) [FactorDesignator]
                    ) [TermFactor]
                  ) [OnlyTerm]
                ) [ExpressionTermDef]
              ) [LeftDesignatorAssignExpression]
            ) [DesignatorStatementStatement]
          ) [StatementListStatement]
          DesignatorStatementStatement(
            LeftDesignatorAssignExpression(
              LeftSideAssignVar(
                DesignatorOnly(
                 niz
                ) [DesignatorOnly]
              ) [LeftSideAssignVar]
              AssignOpEqual(
                AssignOp(
                ) [AssignOp]
              ) [AssignOpEqual]
              ExpressionTermDef(
                OnlyTerm(
                  TermFactor(
                    FactorNewTypeExpression(
                      Type(
                       int
                      ) [Type]
                      ExpressionTermDef(
                        OnlyTerm(
                          TermFactor(
                            FactorConst(
                              NumberConst(
                               5
                              ) [NumberConst]
                            ) [FactorConst]
                          ) [TermFactor]
                        ) [OnlyTerm]
                      ) [ExpressionTermDef]
                    ) [FactorNewTypeExpression]
                  ) [TermFactor]
                ) [OnlyTerm]
              ) [ExpressionTermDef]
            ) [LeftDesignatorAssignExpression]
          ) [DesignatorStatementStatement]
        ) [StatementListStatement]
      ) [MethodDeclarationActual]
    ) [VoidMethodDeclaration]
  ) [MethodDeclarationListMethod]
) [Program]
INFO  01:15:54,775 - ==================SEMANTICKA OBRADA==================
INFO  01:15:54,777 - Deklarisan niz niz(Var niz: array, 0, 0 ) na liniji 7
INFO  01:15:54,777 - Obradjuje se funkcija main(Meth main: none, 0, 0 ) na liniji 9
INFO  01:15:54,778 - Koriscenje  int(int: int, -1, -1 ) na liniji 10
ERROR 01:15:54,778 - Greska na liniji 10 : nije moguca dodela necemusto nije varijabla ili element niza!
INFO  01:15:54,778 - Koriscenje  niz(Var niz: array, 0, 0 ) na liniji 10
INFO  01:15:54,778 - Koriscenje  niz(Var niz: array, 0, 0 ) na liniji 11
ERROR 01:15:54,778 - Greska na liniji 11 : nije moguca dodela2!
INFO  01:15:54,778 - Koriscenje  niz(Var niz: array, 0, 0 ) na liniji 12
INFO  01:15:54,778 - Koriscenje  null(Con null: , 0, 0 ) na liniji 12
INFO  01:15:54,778 - Koriscenje  niz(Var niz: array, 0, 0 ) na liniji 13
INFO  01:15:54,778 - ==================SEMANTICKA ANALIZA==============
INFO  01:15:54,779 - 1  methods in the program
INFO  01:15:54,779 - 0  global variables
INFO  01:15:54,779 - 0  global constants
INFO  01:15:54,779 - 1  global arrays
INFO  01:15:54,779 - 0  local variables in main
INFO  01:15:54,779 - 4  statements in main
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: bool, 0, 1 
Prog test302: notype, 0, 1 
   Var niz: Arr of int, 0, 0 
   Meth main: notype, 0, 0 



ERROR 01:15:54,780 - Program nije semanticki ispravan!
